server/src/main/java/io/crate/planner/node/ddl/AlterTablePlan.java

```diff
@@ -164,10 +165,13 @@ public static Settings.Builder getTableParameter(AlterTable<Object> node, TableP
 
     // Only check for permission if statement is not changing the metadata blocks, so don't block `re-enabling` these.
     static void maybeRaiseBlockedException(TableInfo tableInfo, Settings tableSettings) {
-        if (tableSettings.size() != 1 ||
-            (tableSettings.get(IndexMetadata.SETTING_BLOCKS_METADATA) == null &&
-             tableSettings.get(IndexMetadata.SETTING_READ_ONLY) == null)) {
-
+        Set<String> blockSettings = Set.of(
+            IndexMetadata.SETTING_BLOCKS_METADATA,
+            IndexMetadata.SETTING_READ_ONLY
+        );
+        if (blockSettings.containsAll(tableSettings.keySet())) {
```

Comment:
Wouldn't this then only apply if I set both `blockSettings` together? Shouldn't this also apply if only 1 of these settings are changed? I think the old behavour may also has flaws but it did not raise any exception when one of these blocked settings were detected, while now it will.

Reply:
The `containsAll` behavior if the argument is a set also works if it is only a subset. From the docs:

> Returns `true` if this set contains all of the elements of the specified collection. If the specified collection is also a set, this method returns `true` if it is a *subset* of this set.

```
jshell> Set.of("1", "2", "3").containsAll(Set.of("1"));
$1 ==> true

jshell> Set.of("1", "2", "3").containsAll(Set.of("4"));
$2 ==> false
```

So if it is changing either `blocks.metadata` or `blocks.read_only` or both it will use the `Operation.ALTER_BLOCKS` operation.


Should it also include `blocks.write` and `blocks.read`?

Reply:
Ah sorry, my fault then.

> Should it also include blocks.write and blocks.read?

Good point, probably yes.

Reply:
Also to add some context, without this change `alter table .. reset "blocks.read_only"` failed because with the change to use `putNull` the `tableSettings` both `tableSettings.get(..) == null` checks both passed and it checked for `ALTER` permissions and failed.

Before it worked because one of the `== null` conditions was false due to us adding the default.

Reply:
>> Should it also include blocks.write and blocks.read?
>
> Good point, probably yes.

Done